app/src/renderer/vuex/modules/distribution.js

```diff
@@ -0,0 +1,178 @@
+
+export default ({ node }) => {
+    const emptyState = {
+        loading: false,
+        loaded: false,
+        error: null,
+        /* totalRewards use the following format:
+            { 
+                denom1: amount1,
+                ... ,
+                denomN: amountN 
+            }
+        */
+        totalRewards: {},
+        /* rewards use the following format:
+            { 
+                validatorAddr1: { 
+                    denom1: amount1,
+                    ... ,
+                    denomN: amountN 
+                },
+                ... ,
+                validatorAddrN: { 
+                    denom1: amount1,
+                    ... ,
+                    denomN: amountN 
+                } 
+            } 
+        */
+        rewards: {},
+        parameters: {},
+        /* outstandingRewards use the following format:
+            { 
+                denom1: amount1,
+                ... ,
+                denomN: amountN 
+            }
+        */
+        outstandingRewards: {}
+    }
+    const state = JSON.parse(JSON.stringify(emptyState))
+
+    const mutations = {
+        setTotalRewards(state, rewards) {
+            state.totalRewards = rewards
+        },
+        setDelegationRewards(state, { validatorAddr, rewards }) {
+            Vue.set(state.rewards, validatorAddr, rewards)
+        },
+        setDistributionParameters(state, parameters) {
+            state.parameters = parameters
+        },
+        setOutstandingRewards(state, outstandingRewards) {
+            state.outstandingRewards = outstandingRewards
+        },
+    }
+    const actions = {
+        reconnected({ state, dispatch, rootState: { session } }) {
+            if (state.loading && session.signedIn) {
+                dispatch(`getTotalRewards`)
+            }
+        },
+        resetSessionData({ rootState }) {
+            rootState.delegation = JSON.parse(JSON.stringify(emptyState))
+        },
+        async getTotalRewards(
+            { state, rootState: { session }, commit }
+        ) {
+            state.loading = true
+            try {
+                const rewardsArray = await node.getDelegatorRewards(session.address)
+                const rewards = coinsToObject(rewardsArray)
+                commit(`setTotalRewards`, rewards)
+                state.error = null
+            } catch (error) {
+                commit(`notifyError`, {
+                    title: `Error getting total rewards`,
+                    body: error.message
+                })
+                Sentry.captureException(error)
+                state.error = error
+            }
+            state.loading = false
+            state.loaded = true
+        },
+        async withdrawAllRewards(
+            { rootState: { wallet }, dispatch },
+            { password, submitType }
+        ) {
+            await dispatch(`sendTx`, {
+                type: `postWithdrawDelegatorRewards`,
+                to: wallet.address,
```

Comment:
please always use `session.address` so we don't store the address in 2 places.

Reply:
can we remove `.address` from the wallet module?

Reply:
Yes please

Reply:
#2114